lab_3_ann_retriever/main.py

```diff
@@ -170,6 +215,13 @@ def vectorize(self, tokenized_document: list[str]) -> Vector | None:
 
         In case of corrupt input arguments, None is returned.
         """
+        if not isinstance(tokenized_document, list) or not tokenized_document \
+                or not all(isinstance(word, str) for word in tokenized_document):
+            return None
+        if self._calculate_tf_idf(tokenized_document):
```

Comment:
why are you calling this function twice?

Reply:
where?...